### PR TITLE
update set level()

### PR DESCRIPTION
**Pull request by JaskRendix**
PR addresses:
1. split set_level() into set_level() and set_moves()
2. removing the "learn" component from load_from_db (monster.py)
3. added set_moves close to each set_level -> only exclusion evolving process (only set_level)

Digging deeper I found another issue and the issue was connected to set_level:
`    def set_level(self, level: int = 5) -> None:`
it means that you can have a monster lv 3, but fighting like a lv 5 (related to the #1558 )

Why splitting set_level? Because of the evolution process and max_moves (4)
In this way we have two distinct processes: (1) for levels + stats and (2) for moves 
No interference, no mixing up together.

schematically before
```
-> load_from_db()
--> self.level = random.randint(2, 5) -> 5
--> learn moves (self.level(5)
---> set_level(50) -> learning all the moves not added from lv5 to lv50
----> add the monster
```
schematically after
```
-> load_from_db()
--> set_level(50) 
--> set_moves(50) 
---> add the monster
```
In this way, when we use add_monster or random_monster (eg. a monster lv 30 with 8 moves), it'll come out with the last 4 and not 8.

### CODE REVIEW

File: tuxemon/event/actions/set_monster_level.py [hunk 1]
```diff
@@ -42,7 +52,11 @@ def start(self) -> None:
                 return
 
             monster = self.session.player.monsters[int(monster_slot)]
-            monster.set_level(monster.level + int(monster_level))
+            new_level = monster.level + int(monster_level)
+            monster.set_level(new_level)
+            update_move(monster, new_level)
```
Comment (kerizane) [inline tuxemon/event/actions/set_monster_level.py:57]: Can't we just use the existing monster.update_moves?

**Review by kerizane — CHANGES_REQUESTED** [review]
See comment

File: tuxemon/event/actions/set_monster_level.py (hunk 1 above)
Comment (JaskRendix) [reply tuxemon/event/actions/set_monster_level.py:57]: The existing monster.update_moves? Are you referring to the old piece of code that was inside set_level? I moved it inside set_monster_level
```
            if (
                move.technique not in (m.slug for m in self.moves)
                and move.level_learned <= level
            ):
                self.learn(Technique(move.technique))
```
Do you want that we create a new def update_moves inside monster.py?

Comment (kerizane) [reply tuxemon/event/actions/set_monster_level.py:57]: Can't we use monster.set_moves instead the local function update_move, which is pretty similar?

Comment (JaskRendix) [reply tuxemon/event/actions/set_monster_level.py:57]: Unfortunately we can't. 
Let's say someone uses set_monster_level, so it increases the level of rockitten from 10 to 20, 10 levels
Rockitten knows A, B, C techniques
if we use set_moves, then the result would be:
Rockitten knows A, A, B, B, C, C and D, E, F
(without considering there is the MAX_MOVES parameter in max_moves, so it would be reduced to the last 4, depending on the level)
alternatively we can remove the moves and leaving only the increase of the level, which it would make sense, and I would agree too

File: tuxemon/monster.py [hunk 2]
```diff
@@ -544,13 +541,25 @@ def set_level(self, level: int = 5) -> None:
...
+        Set monster moves according to the level.
+
+        Parameters:
+            level: The level of the monster.
+
+        """
+        moves = []
         for move in self.moveset:
-            if (
-                move.technique not in (m.slug for m in self.moves)
```
Comment (kerizane) [inline tuxemon/monster.py:550]: Why did you move this check into a new method?

**Review by kerizane — CHANGES_REQUESTED** [review]

File: tuxemon/monster.py (hunk 2 above)
Comment (JaskRendix) [reply tuxemon/monster.py:550]: for the same reason I explained in the other comment:
- monster,py (it gives the monster all the techniques from 0 to LEVEL);
- set_monster_level.py (it gives the monster all the techniques from LEVEL to NEW_LEVEL)

Comment (JaskRendix) [reply tuxemon/monster.py:550]: the ex lines 549 and 550 were necessary because there was this (ex lines 317, 318, 319)
```
                if move.level_learned <= self.level:
                    technique = Technique(move.technique)
                    self.learn(technique)
```
we were forced to add that method, now there is no need anymore

distinction:
- monster,py (it gives the monster all the techniques from 0 to LEVEL);
- set_monster_level.py (it gives the monster all the techniques from LEVEL to NEW_LEVEL)

**Review by kerizane — APPROVED** [review]

**Review by kerizane — APPROVED** [review]
